include/odp/api/spec/packet_io.h

```diff
@@ -201,7 +223,22 @@ typedef struct odp_pktin_queue_param_t {
 	  * odp_queue_param_t documentation.
 	  * When classifier is enabled in odp_pktin_queue_config() this
 	  * value is ignored. */
-	odp_queue_param_t queue_param;
+	union {
+		/** Single parameter set for all queues
+		 *
+		 * Coresponds to ODP_PKTIN_QUEUE_PARAM_MODE_BASIC
+		 * queue_param_mode.
+		 */
+		odp_queue_param_t queue_param;
+
+		/** Per queue parameter set
+		 *
+		 * Coresponds to ODP_PKTIN_QUEUE_PARAM_MODE_MULTI
+		 * queue_param_mode.
+		 * queue_params must point to an array with num_queues
+		 * elements. */
+		odp_queue_param_t *queue_params;
```

Comment:
I think this enables too much variability for input queue hashing. Things that are useful to be able set per queue are: schedule group and queue context. It potentially increases implementation complexity if application can request various combinations of other parameters.